language/testing-infra/transactional-test-runner/src/framework.rs

```diff
@@ -140,6 +146,26 @@ pub trait MoveTestAdapter<'a> {
             TaskCommand::Init { .. } => {
                 panic!("The 'init' command is optional. But if used, it must be the first command")
             }
+            TaskCommand::PrintBytecode { .. } => {
+                let state = self.compiled_state();
+                let data = match data {
+                    Some(f) => f,
+                    None => panic!(
+                        "Expected a Move IR module text block following 'print-bytecode' starting on lines {}-{}",
+                        start_line, command_lines_stop
+                    ),
+                };
+                let data_path = data.path().to_str().unwrap();
+                let script = compile_ir_module(state.dep_modules(), data_path)?;
+
+                let source_mapping = SourceMapping::new_from_view(
+                    BinaryIndexedView::Module(&script),
+                    Spanned::unsafe_no_loc(()).loc,
+                )
+                .expect("Unable to build dummy source mapping");
+                let disassembler = Disassembler::new(source_mapping, DisassemblerOptions::new());
+                Ok(Some(disassembler.disassemble()?))
+            }
```

Comment:
In the future, I think we can expand this and allow disassembling of
- scripts in addition to modules
- programs written in the source language

No need to implement this in this PR though.